plugins/modules/source_control/gitlab/gitlab_group_members.py

```diff
@@ -165,98 +201,187 @@ def main():
+        for i in range(len(purge_users)):
+            purge_users[i] = access_level_int[purge_users[i]]
+
     # connect to gitlab server
     gl = gitlabAuthentication(module)
 
     group = GitLabGroup(module, gl)
 
-    gitlab_user_id = group.get_user_id(gitlab_user)
     gitlab_group_id = group.get_group_id(gitlab_group)
 
     # group doesn't exist
     if not gitlab_group_id:
         module.fail_json(msg="group '%s' not found." % gitlab_group)
 
-    # user doesn't exist
-    if not gitlab_user_id:
-        if state == 'absent':
-            module.exit_json(changed=False, result="user '%s' not found, and thus also not part of the group" % gitlab_user)
-        else:
-            module.fail_json(msg="user '%s' not found." % gitlab_user)
-
-    members = group.get_members_in_a_group(gitlab_group_id)
-    is_user_a_member = group.is_user_a_member(members, gitlab_user_id)
-
-    # check if the user is a member in the group
-    if not is_user_a_member:
-        if state == 'present':
-            # add user to the group
-            if not module.check_mode:
-                group.add_member_to_group(gitlab_user_id, gitlab_group_id, access_level)
-            module.exit_json(changed=True, result="Successfully added user '%s' to the group." % gitlab_user)
-        # state as absent
-        else:
-            module.exit_json(changed=False, result="User, '%s', is not a member in the group. No change to report" % gitlab_user)
-    # in case that a user is a member
+    members = []
+    gitlab_users_access = []
+    if module.params['gitlab_user'] is not None:
+        gitlab_users = module.params['gitlab_user']
+        for glu in gitlab_users:
+            gitlab_users_access.append({'name': glu, 'access_level': access_level})
+    elif module.params['gitlab_users_access'] is not None:
+        gitlab_users_access = module.params['gitlab_users_access']
+        for i in range(len(gitlab_users_access)):
+            if gitlab_users_access[i]['access_level']:
+                gitlab_users_access[i]['access_level'] = access_level_int[gitlab_users_access[i]['access_level']]
+
+    if len(gitlab_users_access) == 1:
+        # only single user given
+        members = [group.get_member_in_a_group(gitlab_group_id, group.get_user_id(gitlab_users_access[0]['name']))]
+        if members[0] is None:
+            members = []
+    elif len(gitlab_users_access) > 1 or purge_users:
+        # list of users given
+        members = group.get_members_in_a_group(gitlab_group_id)
     else:
-        if state == 'present':
-            # compare the access level
-            user_access_level = group.get_user_access_level(members, gitlab_user_id)
-            if user_access_level == access_level:
-                module.exit_json(changed=False, result="User, '%s', is already a member in the group. No change to report" % gitlab_user)
+        # something went wrong
+        module.fail_json(msg="Please give at least one user or set purge_users true.")
```

Comment:
I would not fail in this situation, but simply do nothing. After all the user asked the module to do nothing.